src/mx_bluesky/hyperion/external_interaction/agamemnon.py

```diff
@@ -0,0 +1,99 @@
+import dataclasses
+import json
+import re
+from typing import TypeVar
+
+import requests
+from dodal.utils import get_beamline_name
+
+from mx_bluesky.common.parameters.components import WithVisit
+from mx_bluesky.common.parameters.constants import GridscanParamConstants
+from mx_bluesky.common.utils.log import LOGGER
+from mx_bluesky.hyperion.parameters.load_centre_collect import LoadCentreCollect
+
+T = TypeVar("T", bound=WithVisit)
+AGAMEMNON_URL = "http://agamemnon.diamond.ac.uk/"
+MULTIPIN_PREFIX = "multipin"
+MULTIPIN_FORMAT_DESC = "Expected multipin format is multipin_{number_of_wells}x{well_size}+{distance_between_tip_and_first_well}"
+MULTIPIN_REGEX = rf"{MULTIPIN_PREFIX}_(\d+)x(\d*\.?\d*)\+(\d*\.?\d*)"
```

Comment:
Not sure how important this really is, but this regex allows for the following cases:
`"multipin_6x50+"`
`"multipin_6x50+98."`
`"multipin_6x50+.1"`
I've tried to make this stricter, and came up with this:
```suggestion
MULTIPIN_REGEX = rf"^{MULTIPIN_PREFIX}_(\d+)x(\d+(?:\.\d+)?)\+(\d+(?:\.\d+)?)$"
```
which should prevent invalid floats, and also anchors the string. Tuning this is probably never-ending though, so whatever you think is best! I've commented additional tests we can use, if you do use this regex.

Reply:
Love it, thanks!